packages/eds-core-react/src/components/next/Input/Input.tsx

```diff
@@ -0,0 +1,105 @@
+import { forwardRef, useState, InputHTMLAttributes, CSSProperties } from 'react'
+import { OverridableComponent } from '@equinor/eds-utils'
+import type { InputProps } from './Input.types'
+
+export const Input: OverridableComponent<InputProps, HTMLInputElement> =
+  forwardRef<
+    HTMLInputElement,
+    InputProps & InputHTMLAttributes<HTMLInputElement>
+  >(function Input(
+    {
+      invalid = false,
+      disabled = false,
+      type = 'text',
+      leftAdornments,
+      rightAdornments,
+      readOnly,
+      className,
+      style,
+      leftAdornmentsProps,
+      rightAdornmentsProps,
+      leftAdornmentsWidth,
+      rightAdornmentsWidth,
+      as: Component = 'input',
+      ...inputProps
+    },
+    ref,
+  ) {
+    const [leftAdornmentsRef, setLeftAdornmentsRef] = useState<HTMLDivElement>()
+    const [rightAdornmentsRef, setRightAdornmentsRef] =
+      useState<HTMLDivElement>()
+
+    // Calculate dynamic padding based on adornment widths
+    // Use manual width if provided, otherwise auto-measure from DOM
+    const leftWidth =
+      leftAdornmentsWidth ?? (leftAdornmentsRef?.clientWidth || 0)
+    const rightWidth =
+      rightAdornmentsWidth ?? (rightAdornmentsRef?.clientWidth || 0)
```

Comment:
I would keep things simple. This adds some complexity that I am not sure we actually want/need.